test/github_pull_requests_test.rb

```diff
@@ -244,6 +244,20 @@ def test_pull_request_nothing_happened
     assert_equal({ ok: false, message: "Nothing happened" }, response)
   end
 
+  def test_different_base_url
+    old_base = ENV["CODECLIMATE_GITHUB_API_URL"]
+    ENV["CODECLIMATE_GITHUB_API_URL"] = "http://example.com"
+
+    @stubs.get("/user") do |env|
+      assert env[:url].to_s == "http://example.com/user"
+      [200, { "x-oauth-scopes" => "gist, user, repo" }, ""]
+    end
+
+    assert receive_test({ add_comment: true })[:ok], "Expected test of pull request to be true"
+  ensure
+    ENV["CODECLIMATE_GITHUB_API_URL"] = old_base
```

Comment:
I just created a `stub_env` in `config_spec.rb`, might be worth moving to a shared helper to reuse here?